File: annotation/src/main/java/ru/bachev/annotation/JavaConfigApp.java
package ru.bachev.annotation;

import org.springframework.context.annotation.AnnotationConfigApplicationContext;


public class JavaConfigApp {
    public static void main(String[] args) {
        AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(JavaConfigSpring.class);
        Car car = context.getBean("audi", Car.class);
        System.out.println(car.getModel());
        System.out.println(car.getSpeed());
        System.out.println(car.getEngine());
        context.close();
    }
}

File: annotation/src/main/kotlin/ru/bachev/annotation/JavaConfigApp.kt
package ru.bachev.annotation

import org.springframework.context.annotation.AnnotationConfigApplicationContext
import org.springframework.context.support.ClassPathXmlApplicationContext

fun main() {
    val context = AnnotationConfigApplicationContext(JavaConfigSpring::class.java)
    val car = context.getBean("audi", Car::class.java)
    println(car.model)
    println(car.speed)
    println(car.engine)
    context.close()
}